### ISSUE TITLE
Scaling images rarely works.

### ISSUE THREAD

**Issue by abrunyate** [wagtail/wagtail#5071]
### Issue Summary

Scaling of images doesn't work if the scaling factor produces a non-integral dimension.

### Steps to Reproduce

Render a template of the form
  `{% image my_image scale-50 as my_rendition %}`
for an image with odd height or width. Pillow will generate an exception from being passed non-integral dimensions. PR forthcoming. This was tested with a custom image class, but should occur regardless.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (no)

### Technical details

* Python version: 3.6.3
* Django version: 2.1.5
* Wagtail version: 2.5
* Browser version: Chromium 65

**Comment by abrunyate** [comment]
Odd. The tests explicitly allow this, so *someone* thought it would work. My Pillow version is 5.4.1, which seems current. I'll hold off on submitting the PR to give knowledgeable folks time to comment.

**Comment by lb-** [comment]
@abrunyate  - this is definitely a bug by the looks of it, it looks like the way we mock the usage of the underlying Willow/PIL library means that current tests do not pick this up.

If you are able to, we would love a PR for this - see some more details / notes below.

---

The willow docs state:
> Stretches the image to fit the specified size. Size must be a sequence of two integers:
https://willow.readthedocs.io/en/latest/reference.html#resize

A potential fix would be to use `int` instead of `float` in the image_operations file, at the class `ScaleOperation` OR use `value.round()` as used when the value is passed in to `willow.crop()`. That appears to be the only case of float values being passed in directly to Willow in that file.
https://github.com/wagtail/wagtail/blob/master/wagtail/images/image_operations.py#L228-L229

The test class `WillowOperationRecorder` will need to be updated also to reflect the expected API. The recorder checks if the operation is 'resize' and should probably throw an error if the value received is not an int. Bonus points if we can add checks to the `crop` also in that code, as the docs state these should also be integers. The code that calls willow.crop is rounded but good to have this validation in the tests also.

https://github.com/wagtail/wagtail/blob/master/wagtail/images/tests/test_image_operations.py#L35

This is likely a legacy issue not yet picked up as a result of the Python 3 changes.

---

I did the same change mentioned above on Wagtail Bakery Demo and got the same error.

```

Request Method: | GET
-- | --
http://localhost:8000/
2.1.5
TypeError
integer argument expected, got float
/home/vagrant/.virtualenvs/bakerydemo/lib/python3.5/site-packages/PIL/Image.py in resize, line 1556
/home/vagrant/.virtualenvs/bakerydemo/bin/python
3.5.2
['/vagrant/bakerydemo',  '/vagrant/bakerydemo',  '/home/vagrant/.virtualenvs/bakerydemo/lib/python35.zip',  '/home/vagrant/.virtualenvs/bakerydemo/lib/python3.5',  '/home/vagrant/.virtualenvs/bakerydemo/lib/python3.5/plat-i386-linux-gnu',  '/home/vagrant/.virtualenvs/bakerydemo/lib/python3.5/lib-dynload',  '/usr/lib/python3.5',  '/usr/lib/python3.5/plat-i386-linux-gnu',  '/home/vagrant/.virtualenvs/bakerydemo/lib/python3.5/site-packages',  '/vagrant/wagtail',  '/vagrant/libs/django-modelcluster',  '/vagrant/libs/Willow']
Sun, 17 Mar 2019 06:06:01 +0000
```

**Comment by lb-** [comment]
Quick POC of adding some errors to flag incorrect values passed in to Willow

File: wagtail/images/tests/test_image_operations.py

```python
class WillowOperationRecorder:
    # ... __init__ etc
    def __getattr__(self, attr):
        def operation(*args, **kwargs):
            if attr is 'resize':
                width = args[0][0]
                height = args[0][1]
                if not isinstance(width, int):
                    raise TypeError('Received a width value that is not an integer, Willow expects integer.', width)
                if not isinstance(height, int):
                    raise TypeError('Received a height value that is not an integer, Willow expects integer.', height)
            self.ran_operations.append((attr, args, kwargs))
            return self
```

**Comment by abrunyate** [comment]
Could you take a look at what I sketched up a while ago in PR #5076 ? I haven't looked at this in a while, but I can probably work up something if I know exactly what's required.

**Comment by lb-** [comment]
@abrunyate  - I will put some comments on the PR discussion

**Comment by lb-** [comment]
Has been fixed and merged #5076
